qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/ProviderSteps.java

```diff
@@ -0,0 +1,125 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.contrib.qaframework.automation;
+
+import static org.junit.Assert.assertTrue;
+
+import io.cucumber.java.After;
+import io.cucumber.java.Before;
+import io.cucumber.java.en.And;
+import io.cucumber.java.en.Given;
+import io.cucumber.java.en.Then;
+import io.cucumber.java.en.When;
+
+import org.openmrs.contrib.qaframework.RunTest;
+import org.openmrs.contrib.qaframework.helper.RestClient;
+import org.openmrs.contrib.qaframework.helper.TestData;
+import org.openmrs.contrib.qaframework.page.AdministrationPage;
+import org.openmrs.contrib.qaframework.page.ManageProviderPage;
+import org.openmrs.contrib.qaframework.page.ProviderPage;
+
+public class ProviderSteps extends Steps {
+	
+    private TestData.PersonInfo person;
+    private String personUuid;
+    private String providerUuid;
+    private TestData.TestProvider provider;
+    private AdministrationPage administrationPage;
+    private ManageProviderPage manageProviderPage;
+    private ProviderPage providerPage;
+	
+    @Before(RunTest.HOOK.SELENIUM_PROVIDER)
+    public void setUp() {
+    	initiateWithLogin();
+    	person = TestData.generateRandomPerson();
+    	personUuid = TestData.createPerson(this.person);
+    	TestData.createPerson(this.person);
+    	provider = new TestData.TestProvider(person.uuid, person.uuid);
+    	providerUuid = provider.create();
+        providerUuid = new TestData.TestProvider(personUuid, personUuid).create();
+    }
+    
+    @After(RunTest.HOOK.SELENIUM_PROVIDER)
+    public void tearDown() {
+        RestClient.delete("provider/" + providerUuid, true);
+	quit();
+    }
+    
+    @Given("a user clicks on the administration app")
+    public void launchSystemAdministrationPage() {
+    	administrationPage = homePage.goToAdministration(); 	
+    }
+    
+    @Then ("system loads administration page")
+    public void systemLoadsAdministrationPage() {
+    	assertTrue(textExists("Administration"));
+    }
+    
+    @And ("a user clicks on the manage provider link")
+    public void launchAdvancedAdministrationPage() {
```

Comment:
take a close look at the method name and find out whether it reflects what the method is doing.